src/System.Linq/tests/ShortCircuitingTests.cs

```diff
@@ -57,7 +57,10 @@ public void ListLastDoesntCheckAll()
             var source = Enumerable.Range(0, 10).ToList();
             var pred = new CountedFunction<int, bool>(i => i < 7);
             Assert.Equal(6, source.Last(pred.Func));
-            Assert.Equal(4, pred.Calls);
+
+            // .NET core shortcircuits as an optimization.
+            // See https://github.com/dotnet/corefx/pull/2350.
+            Assert.Equal(PlatformDetection.IsFullFramework ? 10 : 4, pred.Calls);
```

Comment:
Yes, we are ok. There is no expectation that `Last` needs to traverse the whole collection.

---
In reply to: [109671461](https://github.com/dotnet/corefx/pull/17877#discussion_r109671461) [](ancestors = 109671461)